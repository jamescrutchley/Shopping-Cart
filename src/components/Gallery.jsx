import Product from "./Product";

const Gallery = () => {
    return (
        <section className="gallery">
            <Product />
        </section>
    )
}

export default Gallery;