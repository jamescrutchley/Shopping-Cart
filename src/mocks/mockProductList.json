[
    {
      "id": 1,
      "name": "New Product 1",
      "price": 9.49,
      "url": "https://example.com/new-product1"
    },
    {
      "id": 2,
      "name": "New Product 2",
      "price": 14.99,
      "url": "https://example.com/new-product2"
    },
    {
      "id": 3,
      "name": "New Product 3",
      "price": 6.49,
      "url": "https://example.com/new-product3"
    },
    {
      "id": 11,
      "name": "Product 11",
      "price": 21.99,
      "url": "https://example.com/product11"
    },
    {
      "id": 12,
      "name": "Product 12",
      "price": 33.49,
      "url": "https://example.com/product12"
    },
    {
      "id": 13,
      "name": "Product 13",
      "price": 8.99,
      "url": "https://example.com/product13"
    },
    {
      "id": 4,
      "name": "New Product 4",
      "price": 25.99,
      "url": "https://example.com/new-product4"
    },
    {
      "id": 15,
      "name": "Product 15",
      "price": 49.99,
      "url": "https://example.com/product15"
    },
    {
      "id": 9,
      "name": "New Product 9",
      "price": 16.99,
      "url": "https://example.com/new-product9"
    },
    {
      "id": 16,
      "name": "Product 16",
      "price": 72.99,
      "url": "https://example.com/product16"
    }
  ]
  