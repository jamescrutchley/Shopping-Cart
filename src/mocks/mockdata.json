[
    {
      "id": 1,
      "name": "Product 1",
      "price": 19.99,
      "url": "https://example.com/product1"
    },
    {
      "id": 2,
      "name": "Product 2 aaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
      "price": 29.99,
      "url": "https://example.com/product2"
    }
]