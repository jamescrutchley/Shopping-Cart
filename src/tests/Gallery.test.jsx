import { fireEvent, render, screen } from "@testing-library/react";

import { describe, it, expect, vitest, vi, beforeEach } from "vitest";
import { fakeStore } from "../mocks/copiedObj";
import Gallery from '../components/Gallery.jsx'
import { renderWithCartContext, mockCartContext } from "./mockContext/mockContext";


describe('Gallery functions correctly', () => {

    it('gallery renders correctly', () => {
        const { asFragment } = renderWithCartContext(
            <Gallery />,
            mockCartContext
          );
          expect(asFragment()).toMatchSnapshot();
      
    })
})