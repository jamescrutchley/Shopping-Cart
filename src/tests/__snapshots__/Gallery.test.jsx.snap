// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Gallery functions correctly > gallery renders correctly 1`] = `
<DocumentFragment>
  <div
    class="_wrapper_556691"
  >
    <div
      class="_galleryHeaderWrapper_556691"
    >
      <div
        class="_categoryContainer_556691"
      >
        <button
          class="_active_556691"
        >
          All Categories
        </button>
        <button>
          Electronics
        </button>
        <button>
          Jewelry
        </button>
      </div>
      <div
        class="_searchAndSort_556691"
      >
        <div
          class="_sortByContainer_556691"
        >
          <label
            for="sortby"
          >
            Sort By:
          </label>
          <select
            id="sortby"
            name="sortby"
          >
            <option
              default=""
              disabled=""
              value="recommended"
            >
              Recommended
            </option>
            <option
              value="asc"
            >
              Price Ascending
            </option>
            <option
              value="desc"
            >
              Price Descending
            </option>
            <option
              disabled=""
              value="rating"
            >
              Rating
            </option>
            <option
              disabled=""
              value="alpha"
            >
              Alphabetical A-Z
            </option>
          </select>
        </div>
        <div
          class="_searchContainer_777731"
        >
          <label
            for="search"
          />
          <input
            class="_searchInput_777731"
            id="search"
            type="text"
            value=""
          />
          <button
            class="_searchButton_777731"
          >
            <img
              alt="search products"
              src="/src/assets/icons/search.svg"
            />
          </button>
        </div>
      </div>
    </div>
    <section
      class="_gallery_556691"
    >
      <p>
        Loading...
      </p>
      <div
        class="_loadWrapper_556691"
      >
        <div
          class="_showingItemsContainer_556691"
        >
          <p>
            Showing 0 of 0 items
          </p>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
